### PR TITLE
Attempt to use a keytree serialization for rbc to help performance

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

File: src/hbbft_acs.erl [hunk 1]
```diff
@@ -308,12 +308,16 @@ deser_bba_key(AK) ->
...
+    #{data => hbbft_rbc:serialize(RBCData)};
 serialize_rbc_state(#rbc_state{rbc_data=RBCData, result=Result}) ->
-    term_to_binary(#rbc_serialized_state{rbc_data=RBCData, result=Result},
-                   [compressed]).
+    #{result => Result, data => hbbft_rbc:serialize(RBCData)}.
 
 -spec deserialize_rbc_state(rbc_serialized_state() | #{}) -> rbc_state().
+deserialize_rbc_state(#{ data := RBCData}=Data) ->
+    %% new RBC sub serialization
+    #rbc_state{rbc_data=RBCData, result=maps:get(result, Data, undefined)};
```
Comment (jeffgrunewald) [inline src/hbbft_acs.erl:320]: if these are all accepting the ``#rbc_serialized_state{}` why do we need to call `hbbft_rbc:serialize/1` going in but we aren't calling `hbbft_rbc:deserialize/1` coming back out from the rbc_serialized_state to the rbc_state?

File: src/hbbft_acs.erl [hunk 2]
```diff
@@ -308,12 +308,16 @@ deser_bba_key(AK) ->
...
-                   [compressed]).
+    #{result => Result, data => hbbft_rbc:serialize(RBCData)}.
 
 -spec deserialize_rbc_state(rbc_serialized_state() | #{}) -> rbc_state().
+deserialize_rbc_state(#{ data := RBCData}=Data) ->
+    %% new RBC sub serialization
+    #rbc_state{rbc_data=hbbft_rbc:deserialize(RBCData), result=maps:get(result, Data, undefined)};
 deserialize_rbc_state(#rbc_serialized_state{rbc_data=RBCData, result=Result}) ->
     #rbc_state{rbc_data=RBCData, result=Result};
 deserialize_rbc_state(BinRec) ->
```
Comment (jeffgrunewald) [inline src/hbbft_acs.erl:323]:
```suggestion
deserialize_rbc_state(BinRec) when is_binary(BinRec) ->
```
this be a useful guard here?

File: rebar.lock [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{"1.2.0",
+{"1.1.0",
```
Comment (vihu) [inline rebar.lock:1]: what happened here?

**Review by evanmcc — APPROVED** [review]
revert the lock and config changes and I think this is ready

**Review by vihu — APPROVED** [review]

**Pull request merged**